### PR TITLE
Update `gt4py` to `v36` and add skip passes

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Purpose

This PR implements the `skip_passes` option in `gt4py` in the `FrozenStencil` decorator class of `fv3core`. This feature is necessary to enable all tests to pass with the `gtc:gt:gpu` and `gtc:cuda` backends in the `v36` version of `gt4py-develop`.

## Code changes:

- Add `skip_passes` option to `FrozenStencil.__init__`
- The `graph_merge_horizontal_executions` (DaCe) pass is skipped by default
- Skip the `KCacheDetection` pass in the `FVSubgridZ.init` stencil
- Remove unused `_zh_tmp` field in `UpdateDzD`
- Skip the `GreedyMerging` pass in `YTP_V._ytp_v` stencil
- Modify `future_stencil` decorator to pass `device_sync` and `skip_passes` to `gt4py` as backend options
- Add `global_config.is_gtc_backend` method

## Requirements changes:

- The `GT4PY_VERSION` is updated to `v36`

## Infrastructure changes:

- The `gt4py_tests_gpu` target of the `Makefile` is modified to skip legacy gridtools v1 backend tests 

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate

### CODE REVIEW

**Comment by ofuhrer** [comment]
Can one of the admins verify this patch?

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by twicki** [comment]
ok to test

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

File: fv3core/decorators.py [hunk 1]
```diff
@@ -103,6 +104,7 @@ def __init__(
             domain: gt4py domain to use at call time
             stencil_config: container for stencil configuration
             externals: compile-time external variables required by stencil
+            skip_passes: compiler passes to skip when building stencil
```
Comment (FlorianDeconinck) [inline fv3core/decorators.py:107]: Can we get a comment here to note that this should be a temporary solution has we introduce a config system

File: fv3core/stencils/updatedzd.py [hunk 2]
```diff
@@ -279,9 +279,6 @@ def _allocate_temporary_storages(self, grid_indexing):
         self._fy = utils.make_storage_from_shape(
             largest_possible_shape, grid_indexing.origin_full()
         )
-        self._zh_tmp = utils.make_storage_from_shape(
```
Comment (FlorianDeconinck) [inline fv3core/stencils/updatedzd.py:282]: Nice catch

**Review by FlorianDeconinck — APPROVED** [review]
LGTM

**Comment by eddie-c-davis** [comment]
launch jenkins

**Pull request merged**